Machine learning for disaster response blog post

merveenoyan opened the pull request:
Pinging @alaradirik for proof-reading.

alaradirik commented on using-ml-for-disasters.md:
```diff
@@ -0,0 +1,81 @@
+<!-- {blog_metadata} -->
+<!-- {authors} -->
+
+On February 6, 2023, earthquakes measuring 7.7 and 7.6 hit the South Eastern Turkey, affecting 10 cities and resulting in more than 42.000 deaths and 120.000 injuries as of February 21.
```
```suggestion
On February 6, 2023, earthquakes measuring 7.7 and 7.6 hit South Eastern Turkey, affecting 10 cities and resulting in more than 42.000 deaths and 120.000 injuries as of February 21.
```

alaradirik commented on using-ml-for-disasters.md:
```diff
@@ -0,0 +1,81 @@
+
+On February 6, 2023, earthquakes measuring 7.7 and 7.6 hit the South Eastern Turkey, affecting 10 cities and resulting in more than 42.000 deaths and 120.000 injuries as of February 21.
+
+In few hours after the earthquake, a group of programmers started a Discord server to roll out an application called *afetharita*, literally meaning, *disaster map*. This application would serve search and rescue teams, and volunteers to find people and bring them their needs. The need for such app arose from how people posted Instagram screenshots of texts where they posted their address and wha ...
```
Correcting minor typos:

```suggestion
In few hours after the earthquake, a group of programmers started a Discord server to roll out an application called *afetharita*, literally meaning, *disaster map*. This application would serve search and rescue teams, and volunteers to find people and bring them their needs. The need for such an app arose from how people posted Instagram screenshots of texts where they posted their addresses and what they needed (including rescue). They also posted tweets of such screenshots or just tweeted what they need so other people knew they are alive. Needing to extract information from these tweets, we developed various applications to turn them into structured data and raced against time in developing and deploying these apps. 
```

alaradirik commented on using-ml-for-disasters.md:
```diff
@@ -0,0 +1,81 @@
+
+In few hours after the earthquake, a group of programmers started a Discord server to roll out an application called *afetharita*, literally meaning, *disaster map*. This application would serve search and rescue teams, and volunteers to find people and bring them their needs. The need for such app arose from how people posted Instagram screenshots of texts where they posted their address and wha ...
+
+When I got invited to discord server, there was quite a lot of chaos on how we would operate and what we would do. The soundest thing would be to open a Hugging Face organization account and collaborate through pull requests as we’ve decided we’d like to build ML based applications to receive and process information and also we needed an open registry for models and datasets. 
```
```suggestion
When I got invited to the discord server, there was quite a lot of chaos regarding how we would operate and what we would do. The soundest thing would be to open a Hugging Face organization account and collaborate through pull requests as we’ve decided we’d like to build ML-based applications to receive and process information and also we needed an open registry for models and datasets. 
```

alaradirik commented on using-ml-for-disasters.md:
```diff
@@ -0,0 +1,81 @@
+
+![organization](assets/using-ml-for-disasters/org.png)
+
+On the first day, people were in need of an application to post screenshots, extract information from these and write the structured information to the database, so we started developing an application that would take a given image, extract a name, telephone number and address and would write it to a database that would be handed to authorities. After experimenting with various open-source OCR to ...
```
Minor corrections and suggestions:
```suggestion
On the first day, people need an application to post screenshots, extract information from these and write the structured information to the database, so we started developing an application that would take a given image, extract a name, telephone number, and address and would write it to a database that would be handed to authorities. After experimenting with various open-source OCR tools, we started using `easyocr` and `Gradio` to develop this application and opened endpoints from the interface so that any team could benefit from OCR as this was the specific need. As of now, the addresses are parsed using transformers-based fine-tuned NER model. 
```

alaradirik commented on using-ml-for-disasters.md:
```diff
@@ -0,0 +1,81 @@
+
+![NER](assets/using-ml-for-disasters/deprem-ner.png)
+
+The model was later used in `afetharita` to parse the addresses. Later, the parsed addresses would hit geocoding API, longitude and latitude would be derived and then passed main backend that would later hand these out to be shown in front-end. Using Inference API for this has saved us from pulling the model, writing an app, building a docker image, setting up CI/CD and deploying the model to a c ...
```
```suggestion
The model was later used in `afetharita` to parse the addresses. The parsed addresses would be sent to the geocoding API to parse the longitude and latitude, and the parsed geolocation would then be displayed on the front-end map. Using Inference API for this has saved us from pulling the model, writing an app, building a docker image, setting up CI/CD, and deploying the model to a cloud instance, where it would be extra overhead work for the DevOps and cloud teams as well. Hugging Face teams have provided us with more replicas so that there would be no downtime and the application would be robust against a lot of traffic.
```

alaradirik commented on using-ml-for-disasters.md:
```diff
@@ -0,0 +1,81 @@
+
+![backend_pipeline](assets/using-ml-for-disasters/production_pipeline.png)
+
+Later on, we were asked to create an intent classification model to extract needs of the survivors, or rescue them. We were given data with multiple labels for multiple needs in a given tweet, and these needs could be shelter, food or logistics, as it was freezing cold over there. We’ve started experimenting first with zero-shot experimentations with open-source NLI models on Hugging Face Hub and ...
```
```suggestion
Later on, we were asked to create an intent classification model to extract the needs of the survivors or rescue them. We were given data with multiple labels for multiple needs in a given tweet, and these needs could be shelter, food, or logistics, as it was freezing cold over there. We’ve started experimenting first with zero-shot experimentations with open-source NLI models on Hugging Face Hub and few-shot experimentations with closed-source generative model endpoints. NLI models were particularly useful as we could directly infer with candidate labels and change the labels as data drift occurs, whereas generative models could have made up labels and  cause mismatches when giving responses to the backend. We initially didn’t have labeled data so anything would work.
```

alaradirik commented on using-ml-for-disasters.md:
```diff
@@ -0,0 +1,81 @@
+
+Later on, we were asked to create an intent classification model to extract needs of the survivors, or rescue them. We were given data with multiple labels for multiple needs in a given tweet, and these needs could be shelter, food or logistics, as it was freezing cold over there. We’ve started experimenting first with zero-shot experimentations with open-source NLI models on Hugging Face Hub and ...
+
+In the end, we decided to fine-tune our own model as it would take roughly three minutes to fine-tune BERT’s text classification head on GPU. We had a separate labelling effort to develop our own model. We noted down our experiments in model card’s metadata so we could later come up with a leaderboard to keep track of which model should be deployed to production. 
```
```suggestion
In the end, we decided to fine-tune our own model as it would take roughly three minutes to fine-tune BERT’s text classification head on a single GPU. We had a separate labeling effort to develop our own model. We logged our experiments in the model card’s metadata so we could later come up with a leaderboard to keep track of which model should be deployed to production. 
```

alaradirik commented on using-ml-for-disasters.md:
```diff
@@ -0,0 +1,81 @@
+
+![intent_model](assets/using-ml-for-disasters/model-repo.png)
+
+We cared a lot about not having false negatives and our classes were imbalanced so we had a benchmark over recall and F1-score hosted in a Space. We were setting metadata tags to be `deprem-clf-v1` and write down F1 and recalls so that Space would automatically retrieve and rank them. We had a separate benchmark set to avoid leakage to train set and consistently benchmark our models. We’ve done b ...
```
```suggestion
Considering the task in hand and the imbalance of our data classes, we focused on eliminating false negatives and and created a Space to benchmark the recall and F1-scores of all models. To do this, we added the metadata tag `deprem-clf-v1` to all relevant model repos and used this tag to automatically retrieve the logged F1 and recall scores and rank models. We had a separate benchmark set to avoid leakage to the train set and consistently benchmark our models. We’ve also benchmarked each model to identify the best threshold per label for deployment.
```

alaradirik commented on using-ml-for-disasters.md:
```diff
@@ -0,0 +1,81 @@
+
+![leaderboard](assets/using-ml-for-disasters/leaderboard.png)
+
+We wanted our NER model to be evaluated and crowd-source the effort because the data labelers were working to give us better and updated intent datasets. To evaluate the NER model, we’ve setup a labelling interface using `Argilla` and `Gradio`, where people could input a tweet and flag the output as correct/incorrect/ambiguous.
```
```suggestion
We wanted our NER model to be evaluated and crowd-source the effort because the data labelers were working to give us better and updated intent datasets. To evaluate the NER model, we’ve set up a labeling interface using `Argilla` and `Gradio`, where people could input a tweet and flag the output as correct/incorrect/ambiguous.
```

alaradirik commented on using-ml-for-disasters.md:
```diff
@@ -0,0 +1,81 @@
+
+![active_learning](assets/using-ml-for-disasters/active-learning.png)
+
+Later, the dataset were deduplicated and we decided to use the dataset to benchmark our further experiments. 
```
```suggestion
Later, the dataset was deduplicated and used to benchmark our further experiments. 
```

alaradirik commented on using-ml-for-disasters.md:
```diff
@@ -0,0 +1,81 @@
+
+Later, the dataset were deduplicated and we decided to use the dataset to benchmark our further experiments. 
+
+Another team under machine learning has worked with generative models (behind a gated API) to get the specific needs (as labels were too broad) as free text and pass the text as an additional context to each posting. For this, they’ve done prompt engineering and wrapped the API endpoints as a separate API and deployed on cloud. Using few-shot prompting with LLMs help adjust in fine-grained needs  ...
```
```suggestion
Another team under machine learning has worked with generative models (behind a gated API) to get the specific needs (as labels were too broad) as free text and pass the text as an additional context to each posting. For this, they’ve done prompt engineering and wrapped the API endpoints as a separate API, and deployed on the cloud. We found that using few-shot prompting with LLMs helps adjusting to fine-grained needs in the presence of rapidly developing data drift, as the only thing we need to adjust is the prompt and we do not need any labeled data for this.
```

alaradirik commented on using-ml-for-disasters.md:
```diff
@@ -0,0 +1,81 @@
+
+![afetharita](assets/using-ml-for-disasters/afetharita.png)
+
+We’ve realized that if it wasn’t for Hugging Face Hub and the ecosystem, we wouldn’t be able to collaborate, prototype and deploy this fast. Below is our MLOps pipeline for address recognition and intent classification models. 
```
```suggestion
We’ve realized that if it wasn’t for Hugging Face Hub and the ecosystem, we wouldn’t be able to collaborate, prototype, and deploy this fast. Below is our MLOps pipeline for address recognition and intent classification models. 
```

alaradirik commented on using-ml-for-disasters.md:
```diff
@@ -0,0 +1,81 @@
+
+![mlops](assets/using-ml-for-disasters/pipeline.png)
+
+There’s tens of people behind this application and its individual components, who worked with no sleep to get these out in short time. 
```
```suggestion
There are tens of people behind this application and its individual components, who worked with no sleep to get these out in such a short time. 
```

alaradirik commented on using-ml-for-disasters.md:
```diff
@@ -0,0 +1,81 @@
+
+## Remote Sensing Applications
+
+Other teams worked on remote sensing applications to assess the damage to buildings and infrastructure in an effort to direct search and rescue operations. The lack of electricity and stabile mobile networks during the first 48 hours of the earthquake, combined with collapsed roads, made it extremely difficult to assess the extent of the damage and where help was needed. The search and rescue ope ...
```
```suggestion
Other teams worked on remote sensing applications to assess the damage to buildings and infrastructure in an effort to direct search and rescue operations. The lack of electricity and stable mobile networks during the first 48 hours of the earthquake, combined with collapsed roads, made it extremely difficult to assess the extent of the damage and where help was needed. The search and rescue operations were also heavily affected by false reports of collapsed and damaged buildings due to the difficulties in communication and transportation.
```

alaradirik commented on using-ml-for-disasters.md:
```diff
@@ -0,0 +1,81 @@
+
+Other teams worked on remote sensing applications to assess the damage to buildings and infrastructure in an effort to direct search and rescue operations. The lack of electricity and stabile mobile networks during the first 48 hours of the earthquake, combined with collapsed roads, made it extremely difficult to assess the extent of the damage and where help was needed. The search and rescue ope ...
+
+In an effort to address these issues and create open source tools that can be leveraged in the future, we started by collecting pre and post-earthquake satellite images of the affected zones from Planet Labs, Maxar and Copernicus Open Access Hub. 
```
```suggestion
To address these issues and create open source tools that can be leveraged in the future, we started by collecting pre and post-earthquake satellite images of the affected zones from Planet Labs, Maxar and Copernicus Open Access Hub. 
```

alaradirik commented on using-ml-for-disasters.md:
```diff
@@ -0,0 +1,81 @@
+
+![app](assets/using-ml-for-disasters/app.png)
+
+Once again, dozens of people worked on labeling, preparing data and training models. In addition to individual volunteers, companies like [Co-One](https://co-one.co/) volunteered to label satellite data with more detailed annotations for buildings and infrastructure, including *no damage, destroyed, damaged, damaged facility,* and *undamaged facility* labels*.* Our current objective is to release ...
```
```suggestion
Once again, dozens of people worked on labeling, preparing data, and training models. In addition to individual volunteers, companies like [Co-One](https://co-one.co/) volunteered to label satellite data with more detailed annotations for buildings and infrastructure, including *no damage*, *destroyed*, *damaged*, *damaged facility,* and *undamaged facility* labels. Our current objective is to release an extensive open-source dataset that can expedite search and rescue operations worldwide in the future.
```

alaradirik reviewed (approved):
I left a few minor comments to correct typos but looks great to me otherwise!

pcuenca commented on using-ml-for-disasters.md:
```diff
@@ -0,0 +1,81 @@
+
+![NER](assets/using-ml-for-disasters/deprem-ner.png)
+
+The model was later used in `afetharita` to parse the addresses. The parsed addresses would be sent to the geocoding API to parse the longitude and latitude, and the parsed geolocation would then be displayed on the front-end map. Using Inference API for this has saved us from pulling the model, writing an app, building a docker image, setting up CI/CD, and deploying the model to a cloud instance ...
```
```suggestion
The model was later used in `afetharita` to extract addresses. The parsed addresses would be sent to a geocoding API to obtain longitude and latitude, and the geolocation would then be displayed on the front-end map. Using Inference API for this has saved us from pulling the model, writing an app, building a docker image, setting up CI/CD, and deploying the model to a cloud instance, where it would be extra overhead work for the DevOps and cloud teams as well. Hugging Face teams have provided us with more replicas so that there would be no downtime and the application would be robust against a lot of traffic.
```

What does `this` encompass above? NER extraction + geocoding?

osanseviero replied:
Some additional comments to Pedro's

```suggestion
The model was later used in `afetharita` to extract addresses. The parsed addresses would be sent to a geocoding API to obtain longitude and latitude, and the geolocation would then be displayed on the front-end map. Using the Inference API has saved us from pulling the model, writing an app, building a docker image, setting up CI/CD, and deploying the model to a cloud instance, where it would be extra overhead work for the DevOps and cloud teams as well. The Hugging Face teams provided us with additional replicas so there would be no downtime, and the application would be robust against a lot of traffic.
```

pcuenca commented on using-ml-for-disasters.md:
```diff
@@ -0,0 +1,81 @@
+
+![backend_pipeline](assets/using-ml-for-disasters/production_pipeline.png)
+
+Later on, we were asked to create an intent classification model to extract the needs of the survivors or rescue them. We were given data with multiple labels for multiple needs in a given tweet, and these needs could be shelter, food, or logistics, as it was freezing cold over there. We’ve started experimenting first with zero-shot experimentations with open-source NLI models on Hugging Face Hub ...
```
```suggestion
Later on, we were asked to create an intent classification model to extract the needs of the survivors or rescue them, based on the tweets they wrote. Any given tweet could potentially express multiple intents corresponding to basic needs such as shelter, food, or logistics, as it was freezing cold over there. We started experimenting with zero-shot open-source NLI models on the Hugging Face Hub and also ran few-shot experiments with closed-source generative model endpoints. NLI models were particularly useful as we could directly infer using candidate labels and change the labels as data drift occurred. Generative models, on the other hand, could have made labels up and cause mismatches when sending the generated responses to the backend. We initially didn’t have labeled data so anything would work.
```

(Not sure if I fully understood the task)

osanseviero replied:
I was a bit confused with "we were asked to create an intent classification model to extract the needs of the survivors or rescue them, based on the tweets they wrote.". Which are the classes the model would predict?

osanseviero replied:
Some additional suggestions to Pedro but my question above still holds

```suggestion
Later on, we created an intent classification model to extract the needs of the survivors or rescue them based on the tweets they wrote. Any tweet could express multiple intents corresponding to basic needs such as shelter, food, or logistics. We started experimenting with zero-shot open-source NLI models on the Hugging Face Hub and also ran few-shot experiments with closed-source generative model endpoints. NLI models were particularly useful as we could directly infer using candidate labels and change the labels as data drift occurred. On the other hand, generative models could have made labels up and caused mismatches when sending the generated responses to the backend. We initially didn’t have labeled data, so anything would work.
```

pcuenca commented on using-ml-for-disasters.md:
```diff
@@ -0,0 +1,81 @@
+
+![intent_model](assets/using-ml-for-disasters/model-repo.png)
+
+Considering the task in hand and the imbalance of our data classes, we focused on eliminating false negatives and and created a Space to benchmark the recall and F1-scores of all models. To do this, we added the metadata tag `deprem-clf-v1` to all relevant model repos and used this tag to automatically retrieve the logged F1 and recall scores and rank models. We had a separate benchmark set to av ...
```
```suggestion
Considering the task at hand and the imbalance of our data classes, we focused on eliminating false negatives and created a Space to benchmark the recall and F1-scores of all models. To do this, we added the metadata tag `deprem-clf-v1` to all relevant model repos and used this tag to automatically retrieve the logged F1 and recall scores and rank models. We had a separate benchmark set to avoid leakage to the train set and consistently benchmark our models. We also benchmarked each model to identify the best threshold per label for deployment.
```

pcuenca commented on using-ml-for-disasters.md:
```diff
@@ -0,0 +1,81 @@
+
+![leaderboard](assets/using-ml-for-disasters/leaderboard.png)
+
+We wanted our NER model to be evaluated and crowd-source the effort because the data labelers were working to give us better and updated intent datasets. To evaluate the NER model, we’ve set up a labeling interface using `Argilla` and `Gradio`, where people could input a tweet and flag the output as correct/incorrect/ambiguous.
```
```suggestion
We wanted our NER model to be evaluated and crowd-sourced the effort because the data labelers were working to give us better and updated intent datasets. To evaluate the NER model, we set up a labeling interface using `Argilla` and `Gradio`, where people could input a tweet and flag the output as correct/incorrect/ambiguous.
```

pcuenca commented on using-ml-for-disasters.md:
```diff
@@ -0,0 +1,81 @@
+
+Once again, dozens of people worked on labeling, preparing data, and training models. In addition to individual volunteers, companies like [Co-One](https://co-one.co/) volunteered to label satellite data with more detailed annotations for buildings and infrastructure, including *no damage*, *destroyed*, *damaged*, *damaged facility,* and *undamaged facility* labels. Our current objective is to re ...
+
+![output_satellite](assets/using-ml-for-disasters/output_satellite.png)
```
Maybe end with a conclusion / takeaway?

pcuenca reviewed (commented):
Impressive and inspiring work. Proud to be part of a team where these things can happen. Feel free to ignore any and all suggestions, of course.

osanseviero commented on using-ml-for-disasters.md:
```diff
@@ -0,0 +1,81 @@
+---
```
Except for the thumbnail, all images should go in https://huggingface.co/datasets/huggingface/documentation-images to make this repository lighter

osanseviero commented on using-ml-for-disasters.md:
```diff
@@ -0,0 +1,81 @@
+---
```
There are some assets that are very large (e.g. output_satellite is over 2 MB.). It's good idea to compress them, as otherwise the experience to load the blog can be very slow in some countries or it could be expensive if reading on phone.

osanseviero replied:
Still to be solved. E.g. https://huggingface.co/datasets/huggingface/documentation-images/blob/main/disaster-assets/app.png is 3+ MB

merveenoyan replied:
Just replaced them.

osanseviero commented on using-ml-for-disasters.md:
```diff
@@ -0,0 +1,81 @@
+- user: adirik
+---
+
+## Using Machine Learning to Aid Survivors and Race through Time
```
Having just one `#` means it's a `h1` which is better for indexing, specially as this is the main title

```suggestion
# Using Machine Learning to Aid Survivors and Race through Time
```

osanseviero commented on using-ml-for-disasters.md:
```diff
@@ -0,0 +1,81 @@
+
+On February 6, 2023, earthquakes measuring 7.7 and 7.6 hit South Eastern Turkey, affecting 10 cities and resulting in more than 42.000 deaths and 120.000 injuries as of February 21.
+
+In few hours after the earthquake, a group of programmers started a Discord server to roll out an application called *afetharita*, literally meaning, *disaster map*. This application would serve search and rescue teams, and volunteers to find people and bring them their needs. The need for such an app arose from how people posted Instagram screenshots of texts where they posted their addresses an ...
```
Some additional changes to Pedro's

```suggestion
In few hours after the earthquake, a group of programmers started a Discord server to roll out an application called *afetharita*, literally meaning, *disaster map*. This application would serve search & rescue teams and volunteers to find people and bring them help. The need for such an app arose when people posted screenshots of texts with their addresses and what they needed (including rescue) on social media. Some people also tweeted what they needed so other people knew they were alive. Needing to extract information from these tweets, we developed various applications to turn them into structured data and raced against time in developing and deploying these apps. 
```

osanseviero commented on using-ml-for-disasters.md:
```diff
@@ -0,0 +1,81 @@
+
+On February 6, 2023, earthquakes measuring 7.7 and 7.6 hit South Eastern Turkey, affecting 10 cities and resulting in more than 42,000 deaths and 120,000 injured as of February 21.
+
+In few hours after the earthquake, a group of programmers started a Discord server to roll out an application called *afetharita*, literally meaning, *disaster map*. This application would serve search and rescue teams, and volunteers to find people and bring them their needs. The need for such an app arose from how people posted Instagram screenshots of texts where they posted their addresses an ...
```
The usage of "we" in this blog post can be a bit confusing as it's not clear if it refers to Hugging Face or to you individually or to the community; it might be worth clarifying

osanseviero commented on using-ml-for-disasters.md:
```diff
@@ -0,0 +1,81 @@
+
+![CI-bot](assets/using-ml-for-disasters/ci-bot.png)
+
+Later on, we were given labelled content from various channels that included addresses and personal information (which is later anonymized) of people who would ask for help. We started experimenting both with few-shot prompting of closed source models and training our own token classification model from transformers. We’ve used `dbmdz/bert-base-turkish-cased` as a base model for token classificat ...
```
```suggestion
Later on, we were given labeled content from various channels that included addresses and personal information (which was then anonymized) of people who would ask for help. We started experimenting with few-shot prompting of closed source models and training our token classification model from transformers. We’ve used `dbmdz/bert-base-turkish-cased` as a base model for token classification and came up with the first address extraction model. 
```

osanseviero replied:
Maybe worth linking to the token classification task page when mentioning it as some people might now know what it is

osanseviero commented on using-ml-for-disasters.md:
```diff
@@ -0,0 +1,81 @@
+
+![backend_pipeline](assets/using-ml-for-disasters/production_pipeline.png)
+
+Later on, we were asked to create an intent classification model to extract the needs of the survivors or rescue them. We were given data with multiple labels for multiple needs in a given tweet, and these needs could be shelter, food, or logistics, as it was freezing cold over there. We’ve started experimenting first with zero-shot experimentations with open-source NLI models on Hugging Face Hub ...
```
Would be interesting to read which models you tried out for zero-shot classification

osanseviero commented on using-ml-for-disasters.md:
```diff
@@ -0,0 +1,81 @@
+
+Later on, we were asked to create an intent classification model to extract the needs of the survivors or rescue them. We were given data with multiple labels for multiple needs in a given tweet, and these needs could be shelter, food, or logistics, as it was freezing cold over there. We’ve started experimenting first with zero-shot experimentations with open-source NLI models on Hugging Face Hub ...
+
+In the end, we decided to fine-tune our own model as it would take roughly three minutes to fine-tune BERT’s text classification head on a single GPU. We had a separate labeling effort to develop our own model. We logged our experiments in the model card’s metadata so we could later come up with a leaderboard to keep track of which model should be deployed to production. 
```
"we decided to fine-tune our own model as it would take roughly three minutes to fine-tune BERT’s text classification head on a single GPU."
"We had a separate labeling effort to develop our own model. "

Are these two different projects/models or were they the same?

osanseviero commented on using-ml-for-disasters.md:
```diff
@@ -0,0 +1,81 @@
+
+Later, the dataset was deduplicated and used to benchmark our further experiments. 
+
+Another team under machine learning has worked with generative models (behind a gated API) to get the specific needs (as labels were too broad) as free text and pass the text as an additional context to each posting. For this, they’ve done prompt engineering and wrapped the API endpoints as a separate API, and deployed on the cloud. We found that using few-shot prompting with LLMs helps adjusting ...
```
```suggestion
Another team under machine learning has worked with generative models (behind a gated API) to get the specific needs (as labels were too broad) as free text and pass the text as an additional context to each posting. For this, they’ve done prompt engineering and wrapped the API endpoints as a separate API, and deployed them on the cloud. We found that using few-shot prompting with LLMs helps adjust to fine-grained needs in the presence of rapidly developing data drift, as the only thing we need to adjust is the prompt and we do not need any labeled data for this.
```

osanseviero commented on using-ml-for-disasters.md:
```diff
@@ -0,0 +1,81 @@
+
+Another team under machine learning has worked with generative models (behind a gated API) to get the specific needs (as labels were too broad) as free text and pass the text as an additional context to each posting. For this, they’ve done prompt engineering and wrapped the API endpoints as a separate API, and deployed on the cloud. We found that using few-shot prompting with LLMs helps adjusting ...
+
+These models are currently being used in production to create the points in below heat map so that volunteers and search and rescue teams can bring the needs to survivors. 
```
```suggestion
These models are currently being used in production to create the points in the heat map below so that volunteers and search and rescue teams can bring the needs to survivors. 
```

osanseviero commented on using-ml-for-disasters.md:
```diff
@@ -0,0 +1,81 @@
+
+Other teams worked on remote sensing applications to assess the damage to buildings and infrastructure in an effort to direct search and rescue operations. The lack of electricity and stable mobile networks during the first 48 hours of the earthquake, combined with collapsed roads, made it extremely difficult to assess the extent of the damage and where help was needed. The search and rescue oper ...
+
+To address these issues and create open source tools that can be leveraged in the future, we started by collecting pre and post-earthquake satellite images of the affected zones from Planet Labs, Maxar and Copernicus Open Access Hub. 
```
Sometimes we say open-source/close-source and sometimes open source and close source; I would try to be consistent

osanseviero commented on using-ml-for-disasters.md:
```diff
@@ -0,0 +1,81 @@
+
+![input_satellite](assets/using-ml-for-disasters/input_satellite.jpeg)
+
+Our initial approach was to rapidly label satellite images for object detection and instance segmentation, with a single category for "buildings". The aim was to evaluate the extent of damage by comparing the number of surviving buildings in pre- and post-earthquake images collected from the same area. In order to make it easier to train models, we started by cropping 1080x1080 satellite images i ...
```
It might be worth linking to task pages here as well

osanseviero reviewed (commented):
Very impressive and inspiring to see everything that the community and team were able to do in such a short timeframe. So many interesting components and feats!

merveenoyan commented on using-ml-for-disasters.md:
```diff
@@ -0,0 +1,81 @@
+
+![OCR](assets/using-ml-for-disasters/ocr-app.png)
+
+Hugging Face Hub team has set us up a CI bot to have an ephemeral environment to see how a given PR would effect the Space. 
```
Addressed this.

merveenoyan commented:
I have:
- Corrected subjects (volunteers, survivors and more)
- Removed assets and moved to HF dataset
- Address @osanseviero and @pcuenca's comments on ambiguity & nits

sayakpaul reviewed (approved):
Inspiring! Lucky to be in a workforce that sees the value in this kind of work.

osanseviero commented on using-ml-for-disasters.md:
```diff
@@ -0,0 +1,82 @@
+
+When I got invited to the discord server, there was quite a lot of chaos regarding how we (volunteers) would operate and what we would do. We decided to collaboratively train models so we needed a model and dataset registry. We opened a Hugging Face organization account and collaborated through pull requests as to build ML-based applications to receive and process information.
+
+![organization](https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/disaster-assets/org.png)
```
The assets should be under https://huggingface.co/datasets/huggingface/documentation-images/tree/main/blog ideally

merveenoyan replied:
isn't it better if they're organized?

osanseviero commented on using-ml-for-disasters.md:
```diff
@@ -0,0 +1,82 @@
+
+![backend_pipeline](https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/disaster-assets/production_pipeline.png)
+
+Later on, we were asked if we could extract what earthquake survivors need from a given tweet. We were given data with multiple labels for multiple needs in a given tweet, and these needs could be shelter, food, or logistics, as it was freezing cold over there. We’ve started experimenting first with zero-shot experimentations with open-source NLI models on Hugging Face Hub and few-shot experiment ...
```
Maybe add links to the hub models in their ids

osanseviero reviewed (approved):
Thanks a lot for the changes 🔥 I left some minor last comments, but almost good to merge!

NimaBoscarino commented on _blog.yml:
```diff
@@ -1874,3 +1875,14 @@
     - diffusers
+
+
+- local: ml-disaster
```
Heads up, this needs to be adjusted to `using-ml-for-disasters`, since the link from https://huggingface.co/blog is broken at the moment

NimaBoscarino replied:
Opened an issue: https://github.com/huggingface/blog/pull/889
